Cabal/src/Distribution/Fields/Field.hs

```diff
@@ -42,12 +44,14 @@ import           Prelude ()
 -- | A Cabal-like file consists of a series of fields (@foo: bar@) and sections (@library ...@).
 data Field ann
     = Field   !(Name ann) [FieldLine ann]
+    | Comment !ann !(CommentLine ann)
```

Comment:
Why?

Comments can be extracted separately? See how `cabal-fmt` does  the job. (Also GHC AST is not polluted with comments, they are attached afterwards. That's why there is `ann` in the `Field` structure too).

Reply:
This was just something I wrote for quick prototyping. I'll checkout both `cabal-fmt` and `ghc-exactprint`

Reply:
We should keep in mind that most time this code is hit when solver is traversing dozens if not hundreds of cabal files. The less stuff is done in hot path, the more responsive `cabal-install` is.

The "interactive" usage should not make compromises to the primary function. I'd argue that if it becomes a friction then having two separate parsers is not impossible idea either.

Reply:
Hmm... it's something to ponder; while having a solid base to build on top of is great, having a clean slate to shape the feature to our current needs, while not affecting the older code, is very appealing.